### ISSUE TITLE
docs: middle ware parameter [how to connect mcollective with a middleware]

### ISSUE THREAD

**Issue by fullmooninu** [voxpupuli/puppet-mcollective#210]
```puppet
node puppet {
  class { '::mcollective':
    client           => true,
    middleware       => true,
    middleware_hosts => ['puppet.domain1.net'],
  }
}
```

using this on puppet master to install mcollective and it's not working because it says "middleware" parameter is not valid?

`Invalid parameter middleware on Class[Mcollective] at /etc/puppet/manifests/site.pp:6 on node puppet.salesseek.net`

**Comment by igalic** [comment]
well, it's not entirely wrong, https://github.com/puppet-community/puppet-mcollective/blob/master/manifests/init.pp#L28-L41

**Comment by fullmooninu** [comment]
Was reading that just now :P
So the readme is outdated. What kind of manifest do I need to set up a puppetmaster with the middleware and an agent?

```puppet
node puppet {
  class { '::mcollective': middleware_hosts => ['puppet'], }
}

node puppetagent {
 # class { '::mcollective':
 #   client           => true,
 #   middleware_hosts => ['puppet'],
 # }
}
 ```

This is what I got so far, but getting a dependency errors, so I'm thinking it's not enough?

**Comment by amitgilad3** [comment]
I am also stuck on this issue ,
can anybody explain how to solve it?

**Comment by ffrank** [comment]
Full disclosure, I've never used or hacked this module.

Still, I just went hunting for your missing parameter, and it turns out we don't support middleware management anymore since 9be4fed4.

To setup your queue manager, you should pick an ActiveMQ or RabbitMQ module and add it to the respective profile(s).

**Comment by fullmooninu** [comment]
Hmmm, @ffrank what you recommend, active or rabbit?

**Comment by ffrank** [comment]
I never tried either option, and cannot supply any second hand knowledge either. Sorry.

**Comment by igalic** [comment]
@fullmooninu depends on your scale, i guess.
for a small (we're running <500 machines) setup, ActiveMQ serves us quite well.

**Comment by fullmooninu** [comment]
ty ty both. I ended up manually installing rabbitmq followed by mcollective, using someone's written instructions.

**Comment by ffrank** [comment]
Oh nooo :anguished: 

And thus we fail...

**Comment by igalic** [comment]
[silent sobbing]

**Comment by fullmooninu** [comment]
*pats*

**Comment by ffrank** [comment]
Reopening so that this won't fall under the rug.

We desperately need to update the docs. This is a pretty serious issue from my point of view.

**Comment by jyaworski** [comment]
Hello:

Is this still needed?

**Comment by ffrank** [comment]
I'll try and clean this up.

**Comment by ffrank** [comment]
Ah, I believe this was finally resolved when #178 got merged. Way to go team :)

Closing this for good.
